src/Raven.Client/Documents/Commands/Batches/PutAttachmentCommandData.cs

```diff
@@ -9,11 +10,20 @@ namespace Raven.Client.Documents.Commands.Batches
 {
     public sealed class PutAttachmentCommandData : ICommandData
     {
-        public PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, string changeVector) : this(documentId, name, stream, contentType, changeVector, false)
+        [Obsolete("This constructor is deprecated and will be removed in next RavenDB major version, use the overload with retireAttachmentParameters instead")]
+        public PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, string changeVector)
+            : this(documentId, name, stream, contentType, changeVector, size: null, retireAttachmentParameters: null, hash: null, fromEtl: false)
         {
+            SizeInBytes = stream.Length;
         }
 
-        internal PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, string changeVector, bool fromEtl)
+        public PutAttachmentCommandData(string documentId, string name, Stream stream, string contentType, string changeVector, RetireAttachmentParameters retireAttachmentParameters) 
+            : this(documentId, name, stream, contentType, changeVector, size: null, retireAttachmentParameters: retireAttachmentParameters, hash: null, fromEtl: false)
+        {
+            SizeInBytes = stream.Length;
```

Comment:
This is new, no? 
What about streams that don't expose their `Length` - weren't they supported previously and now not?